test/suite/test_tiered04.py

```diff
@@ -229,13 +256,48 @@ def test_tiered(self):
         # Call flush_tier with its various configuration arguments. It is difficult
         # to force a timeout or lock contention with a unit test. So just test the
         # call for now.
+        #
+        # There have been no data changes nor checkpoints since the last flush_tier with
+        # force, above. We need to sleep a second because if a checkpoint occurred in the
+        # same second the code flushes. The skip statistics should increase and the switched
+        # statistics should stay the same.
+        skip1 = self.get_stat(stat.conn.flush_tier_skipped, None)
+        switch1 = self.get_stat(stat.conn.flush_tier_switched, None)
         self.session.flush_tier('timeout=10')
         self.session.flush_tier('lock_wait=false')
         self.session.flush_tier('sync=off')
         flush += 3
         self.pr("reconfigure get stat")
         calls = self.get_stat(stat.conn.flush_tier, None)
         self.assertEqual(calls, flush)
+        skip2 = self.get_stat(stat.conn.flush_tier_skipped, None)
+        switch2 = self.get_stat(stat.conn.flush_tier_switched, None)
+        self.assertEqual(switch1, switch2)
+        self.assertGreater(skip2, skip1)
+
+        # Test that the checkpoint and flush times work across a connection restart.
+        # Make modifications and then close the connection (which will checkpoint).
+        # Reopen the connection and call flush_tier. Verify this flushes the object.
+        c = self.session.open_cursor(self.uri)
+        c["4"] = "4"
+        self.check(c, 5)
+        c.close()
+        # Manually reopen the connection because the default function above tries to
+        # make the bucket directories.
+        self.close_conn()
+        self.conn = self.wiredtiger_open('.', self.saved_conn)
+        self.session = self.conn.open_session()
```

Comment:
I think you can do replace these lines by:
```
    self.reopen_conn(config = self.saved_conn)
```

Reply:
That is great! I didn't know that. I think there are other tiered tests I wrote that do something similar for the same reason. I'll fix those too.

Reply:
Actually those other tests are doing things to the filesystem while the connection is closed to test certain things so they cannot be fixed, but this test works well with this.